client/src/pages/files/addTags.tsx

```diff
@@ -0,0 +1,198 @@
+import Prose from 'components/Prose';
+import React, {
+  useState,
+  useEffect,
+  KeyboardEvent,
+  useRef,
+  useMemo,
+} from 'react';
+
+import {getUserFiles, updateDocumentTags} from 'lib/api/files';
+import {UserFile} from 'types/UserFile';
+import {Table} from 'semantic-ui-react';
+
+import {useAuth} from 'contexts/auth';
+
+export default function TaggingView() {
+  const {user} = useAuth();
+  const [userFiles, setUserFiles] = useState<Map<string, [UserFile, boolean]>>(
```

Comment:
I think this is trying to put too much state in one variable. I reckon `userFiles` should just be the map of filenames to the files, and a separate useState should be a set of all the filenames that will change in this update.

Reply:
I love that idea. Thanks!